revolv/base/views.py

```diff
@@ -59,9 +60,25 @@ def get_context_data(self, **kwargs):
         context["project_dict"] = project_dict
         context["role"] = self.role or "donor"
 
+        context['category_setter_url'] = reverse('dashboard_category_setter')
+        context['categories'] = Category.objects.all().order_by('title')
+        context['preferred_categories'] = RevolvUserProfile.objects.get(user=self.user).preferred_categories.all()
```

Comment:
In a UserDataMixin, you can just use `self.profile` to get the user's RevolvUserProfile